aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h

```diff
@@ -190,5 +211,160 @@ struct conv_param_t {
       ukernel_type = pytorch_qnnp_ukernel_type_conv;
     }
   }
+
+  /**
+   * @brief Computes the output dimensions given a 2D input.
+   */
+  std::array<size_t, 2> compute_output_dims(
+      std::array<size_t, 2> input_dims) const {
+    std::array<size_t, 2> output_dims;
+    output_dims[0] = compute_output_dimension(input_dims[0],  // width
+                                              pad[1] + pad[3],
+                                              /*adjustment=*/0,
+                                              kernel_dims[0],
+                                              dilation[0],
+                                              subsampling_dims[0],
+                                              /*transpose=*/false);
+    output_dims[1] = compute_output_dimension(input_dims[1],  // height
+                                              pad[0] + pad[2],
+                                              /*adjustment=*/0,
+                                              kernel_dims[1],
+                                              dilation[1],
+                                              subsampling_dims[1],
+                                              /*transpose=*/false);
+    return output_dims;
+  }
+};
+
+struct deconv_param_t {
```

Comment:
The PR title says refactor conv but this adds a new struct for deconv. We aren't using this struct here, why don't we move it to the PR where we actually use it?

Reply:
The whole point of the refactor is to prep for the deconv struct, and add hte parameters. I will rename it